### PR TITLE
Replace useless nullability comments with jetbrains annotation

### PR DESCRIPTION
**Pull request by SilverAndro**
This broke my automated tooling which relies on knowing nullability (specifically the block break events fucked it up). I very much hope you aren't planning to add more of these into the codebase because these are tremendously annoying and I assumed it was some "smart view" mode in my IDE hiding the actual annotation.

### CODE REVIEW

**Comment by apple502j** [comment]
Are they all nullable, or in fact not nullable and commented out for reasons?

**Comment by SilverAndro** [comment]
I know the block ones are nullable at least for sure, since thats what exploded at runtime

**Comment by SilverAndro** [comment]
Item ones were old comments, I have no clue on the entity builder because theres both a requireNonNull and a "if not null", which seems to imply it can be null but only while building? Theres 0 hints at the intent here
image

**Comment by SilverAndro** [comment]
Going to assume its supposed to be nullable internally for building, since
- The field is private, meaning its not public API
- The setter requires the passed value to not be null (only place its interactable in public API)
- The build method makes a check if its not null, because it can be null if the public api consumer doesn't call the set method

Since reading isnt exposed anywhere, it doesn't really matter since it has 0 impact on public API, so the extra safety for maintainability makes sense

File: fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/ItemExtensions.java [hunk 1]
```diff
@@ -19,9 +19,11 @@
 import net.fabricmc.fabric.api.item.v1.CustomDamageHandler;
 import net.fabricmc.fabric.api.item.v1.EquipmentSlotProvider;
 
+import org.jetbrains.annotations.Nullable;
+
 public interface ItemExtensions {
-	/* @Nullable */ EquipmentSlotProvider fabric_getEquipmentSlotProvider();
+	EquipmentSlotProvider fabric_getEquipmentSlotProvider();
```
Comment (apple502j) [inline fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/ItemExtensions.java]: Now that I think about this, it is probably nullable. (`ItemMixin` is missing annotation.)

**Review by apple502j — CHANGES_REQUESTED** [review]
Could you run checkstyle and fix the import sorting issues?

File: fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/FabricItemInternals.java [hunk 2]
```diff
@@ -43,8 +45,8 @@ public static void onBuild(Item.Settings settings, Item item) {
 	}
 
 	public static final class ExtraData {
-		private /* @Nullable */ EquipmentSlotProvider equipmentSlotProvider;
-		private /* @Nullable */ CustomDamageHandler customDamageHandler;
+		private EquipmentSlotProvider equipmentSlotProvider;
```
Comment (apple502j) [inline fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/FabricItemInternals.java]: These too are nullable, I believe this was coded back when nullable annotations were not used?

**Review by apple502j — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**